include/nbl/builtin/hlsl/spirv_intrinsics/core.hlsl

```diff
@@ -105,12 +172,26 @@ void controlBarrier(uint32_t executionScope, uint32_t memoryScope, uint32_t memo
 [[vk::ext_instruction( spv::OpMemoryBarrier )]]
 void memoryBarrier(uint32_t memoryScope, uint32_t memorySemantics);
 
-
 // Add specializations if you need to emit a `ext_capability` (this means that the instruction needs to forward through an `impl::` struct and so on)
 template<class T, class U>
 [[vk::ext_instruction(spv::OpBitcast)]]
 T bitcast(U);
```

Comment:
won't this first template match everything? 

IMHO it should be an `enable_if_t<...,T>` that checks for the exact NOT of the OR of the 3 overloads below.

Or at least get declared afterwards.

Reply:
I thought it will not get picked because the compiler prefers the more specific overload? I have moved it down to the end.

Reply:
AFAIK its "first that matches" not "best match"